climada/engine/impact.py

```diff
@@ -390,6 +388,44 @@ def impact_per_year(self, all_years=True, year_range=None):
             year_set[year] = sum(self.at_event[orig_year == year])
         return year_set
 
+    def impact_at_reg(self, agg_regions=None):
+        """Aggregate impact on given aggregation regions. This method works 
+        only if Impact.imp_mat was stored during the impact calculation.
+
+        Parameters
+        ----------
+        agg_regions : np.array, list, pd.Series
+            It has the same lenght as the number of coordinates points in exposure 
+            and reports what macro-regions these points belong to. If no aggregation
+            regions are passed, the method aggregates impact at the admin_0 level.
+            Default is None.
+
+        Returns
+        -------
+        pd.DataFrame
+        """
+        if self.imp_mat.nnz == 0:
+            LOGGER.warning("No Impact.imp_mat was stored during the impact calculation,"
+                           "thus the impact per region cannot be computed")
+            return None
+
+        if agg_regions is None:
+            LOGGER.warning("Aggregation regions were not specified,"
+                           "impact is aggregated at the admin_0 level")
+            agg_regions = pd.Series(u_coord.get_country_code(self.coord_exp[:,0],
+                                                             self.coord_exp[:,1]))
+        elif not isinstance(agg_regions, pd.Series):
+            agg_regions = pd.Series(agg_regions)
+
+        at_reg_event = np.hstack([
+                self.imp_mat[:, np.where(agg_regions == reg)[0]].sum(1)
+                for reg in agg_regions.unique()
+                ])
+
+        at_reg_event = pd.DataFrame(at_reg_event, columns=agg_regions.unique())
```

Comment:
Maybe use event IDs as indices for this data frame? 🤔

Reply:
agreed, done.